src/main/perf/facets/BenchmarkFacets.java

```diff
@@ -0,0 +1,160 @@
+
+        try {
+            System.out.println("Number of docs: " + s.getIndexReader().numDocs());
+            System.out.println("Number of segments: " + s.getIndexReader().leaves().size());
+            String ssdvFieldName = "address.sortedset";
+            String taxoFieldName = "address.taxonomy";
+
+            System.out.println("Testing facet instanciation...");
+
+            FacetsCollector c = new FacetsCollector();
+            s.search(new MatchAllDocsQuery(), c);
+
+            long createFastTaxoCountsStart = System.nanoTime();
+            Facets taxoFacets = new FastTaxonomyFacetCounts(taxoFieldName, taxoReader, config, c);
+            long createFastTaxoCountsEnd = System.nanoTime();
+            double createFastTaxoCountsTime = (double)(createFastTaxoCountsEnd - createFastTaxoCountsStart) / BILLION;
+
+            // only benchmark count creation
+            SortedSetDocValuesReaderState state = new DefaultSortedSetDocValuesReaderState(s.getIndexReader());
+            long createSSDVReaderStateStart = System.nanoTime();
+            Facets ssdvFacets = new SortedSetDocValuesFacetCounts(state, c);
+            long createSSDVReaderStateEnd = System.nanoTime();
+            double createSSDVReaderStateTime = (double)(createSSDVReaderStateEnd - createSSDVReaderStateStart) / BILLION;
+
+            System.out.println("Time (s) taken to instanciate FastTaxonomyFacetCounts: " + createFastTaxoCountsTime);
+            System.out.println("Time (s) taken to instanciate SSDVFacetCounts: " + createSSDVReaderStateTime);
+            System.out.println("Percent difference: " + getPercentDiff(createFastTaxoCountsTime, createSSDVReaderStateTime) + "%");
+
+            // test getAllDims (there is only 1 dim, address.taxonomy or sortedset.taxonomy)
+            System.out.println("Testing getAllDims...");
+
+            long taxoGetAllDimsStart = System.nanoTime();
+            List<FacetResult> results = taxoFacets.getAllDims(5);
+            long taxoGetAllDimsEnd = System.nanoTime();
+            double taxoGetAllDimsTime = (double) (taxoGetAllDimsEnd - taxoGetAllDimsStart) / BILLION;
+
+            long ssdvGetAllDimsStart = System.nanoTime();
+            results = ssdvFacets.getAllDims(5);
+            long ssdvGetAllDimsEnd = System.nanoTime();
+            double ssdvGetAllDimsTime = (double) (ssdvGetAllDimsEnd - ssdvGetAllDimsStart) / BILLION;
+
+            System.out.println("Time (s) taken to get all dims for taxonomy: " + taxoGetAllDimsTime);
+            System.out.println("Time (s) taken to get all dims for SSDV: " + ssdvGetAllDimsTime);
+            System.out.println("Percent difference: " + getPercentDiff(taxoGetAllDimsTime, ssdvGetAllDimsTime) + "%");
+
+            // test get top children
+            System.out.println("Testing getTopChildren...");
+
+            long taxoGetChildrenStart = System.nanoTime();
+            FacetResult result = taxoFacets.getTopChildren(10, taxoFieldName, "TX");
```

Comment:
We should ideally run the above `MatchAllDocsQuery` many times, pulling the facets after each, and then measure total time divided by N, and also do a few warmup iterations.  But this can come later.